optimum/gptq/quantizer.py

```diff
@@ -0,0 +1,571 @@
+                Also known as act-order.
+            sym (`bool`, *optional*, defaults to `True`):
+                Whether to use symetric quantization.
+            true_sequential (`bool`, *optional*, defaults to `True`):
+                Whether to performing sequential quantization even within a single Transformer block.
+            pack_sequentially (`bool`, *optional*, defaults to `True`):
+                Whether to pack the layer just after it is quantized. If False, we will pack the model at the end.
+            use_cuda_fp16 (`bool`, *optional*, defaults to `True`):
+                Whether or not to use optmized cuda kernel for fp16 model. Need to have model in fp16.
+            model_seqlen (`int`, *optional*, defaults to `None`):
+                The model sequence length
+            block_name_to_quantize (`str`, *optional*, defaults to `None`):
+                The transformers block name to quantize.
+            module_name_preceding_first_block (`str`, *optional*, defaults to `None`):
+                The layers that are preceding the first Transformer block.
+        """
+
+        self.bits = bits
+        self.group_size = group_size
+        self.damp_percent = damp_percent
+        self.desc_act = desc_act
+        self.sym = sym
+        self.true_sequential = true_sequential
+        self.pack_sequentially = pack_sequentially
+        self.use_cuda_fp16 = use_cuda_fp16
+        self.model_seqlen = model_seqlen
+        self.block_name_to_quantize = block_name_to_quantize
+        self.module_name_preceding_first_block = module_name_preceding_first_block
+
+        if self.bits not in [2, 4, 6, 8]:
+            raise ValueError("only support quantize to [2,4,6,8] bits.")
+        if self.group_size != -1 and self.group_size <= 0:
+            raise ValueError("group_size must be greater than 0 or equal to -1")
+        if not (0 < self.damp_percent < 1):
+            raise ValueError("damp_percent must between 0 and 1.")
+        for boolean in [
+            "desc_act",
+            "sym",
+            "true_sequential",
+            "pack_sequentially",
+            "use_cuda_fp16",
+        ]:
+            if not isinstance(getattr(self, boolean), bool):
+                raise ValueError(f"{boolean} must be a float")
+        if self.model_seqlen is not None and not isinstance(self.model_seqlen, int):
+            raise ValueError("model_seqlen must be an int")
+        if self.block_name_to_quantize is not None and not isinstance(self.block_name_to_quantize, str):
+            raise ValueError("block_name_to_quantize must be a string")
+        if self.module_name_preceding_first_block is not None and not isinstance(
+            self.module_name_preceding_first_block, list
+        ):
+            raise ValueError("block_name_to_quantize must be a list of string")
+
+    def to_dict(self):
+        """
+        Return the args in dict format
```

Comment:
```suggestion
        Returns the args in dict format.
```

Let's stick to descriptive style within optimum (as in transformers)
https://github.com/huggingface/doc-builder#writing-documentation-for-hugging-face-libraries
https://google.github.io/styleguide/pyguide.html#383-functions-and-methods